api/v1alpha1/trafficmanagerprofile_types.go

```diff
@@ -34,6 +34,13 @@ type TrafficManagerProfile struct {
 // TrafficManagerProfileSpec defines the desired state of TrafficManagerProfile.
 // For now, only the "Weighted" traffic routing method is supported.
 type TrafficManagerProfileSpec struct {
+	// The name of the resource group to contain the Azure Traffic Manager resource corresponding to this profile.
+	// When this profile is created, updated, or deleted, the corresponding traffic manager with the same name will be created, updated, or deleted
+	// in the specified resource group.
+	// +required
```

Comment:
Required? Do we not provide a default one?

Reply:
We won't provide a default one for now.

Later, we can make this resource group field part of the fleet resource so that all the azure resources will be provisioned in the same resource group for one fleet.

At that time, we can make the field optional.